src/firebase_functions/firestore_fn.py

```diff
@@ -82,9 +87,26 @@ class Event(_core.CloudEvent[_core.T]):
 _C1 = _typing.Callable[[_E1], None]
 _C2 = _typing.Callable[[_E2], None]
 
+AuthType = _typing.Literal["service_account", "api_key", "system",
+                           "unauthenticated", "unknown"]
+
+
+@_dataclass.dataclass(frozen=True)
+class EventWithAuthContext(Event[_core.T]):
```

Comment:
FYI we ended up deciding to call the separate event interface FirestoreAuthEvent in the Node SDK. I guess the equivalent here would be AuthEvent?